src/elements/content-sidebar/hooks/useSidebarMetadataFetcher.ts

```diff
@@ -122,59 +125,52 @@ function useSidebarMetadataFetcher(
         [onApiError],
     );
 
-    const deleteMetadataInstanceSuccessCallback = React.useCallback(
-        (metadataInstance: MetadataTemplateInstance) => {
-            const updatedInstances = templateInstances.filter(
-                templateInstance =>
-                    templateInstance.scope !== metadataInstance.scope &&
-                    templateInstance.templateKey !== metadataInstance.templateKey,
-            );
-            setTemplateInstances(updatedInstances);
-        },
-        [templateInstances],
-    );
-
     const handleDeleteMetadataInstance = React.useCallback(
-        (metadataInstance: MetadataTemplateInstance) => {
+        async (metadataInstance: MetadataTemplateInstance): Promise<void> => {
+            setStatus(STATUS.LOADING);
```

Comment:
If for whatever reason `if` below would return from function we would not change status from `LOADING`. Maybe let's move this `setStatus` after the `if`

Reply:
I agree, we should move this